gossip/src/duplicate_shred_handler.rs

```diff
@@ -261,7 +258,16 @@ mod tests {
             timestamp(), // wallclock
             chunk_size,  // max_size
         )?;
-        Ok(chunks)
+        if let Some(Error::ShredIndexMismatch) = expected_error {
+            Ok(Box::new(chunks.map(|mut duplicate_shred| {
+                if duplicate_shred.chunk_index() > 0 {
+                    duplicate_shred.shred_index += 1
```

Comment:
Looks like all this extra logic is just to ensure chunk index 0's shred indices stay as zero. Can you explain why we want that or what's going on here?

Reply:
previously `ShredIndexMismatch` would be triggered if the shreds did not have the same index. Now that we allow that case for last_shred_in_slot and erasure meta conflicts, instead we can only `ShredIndexMismatch` if the various chunks of the duplicate proof have differing `shred_index` values.